docs/_tutorials/read-only-json-api.md

```diff
@@ -0,0 +1,354 @@
+to see them directly in the browser as `/foo.json`. They can be used to build a custom frontend using Angular, React,
+Vue and so on.
+
+A huge inspiration to write this tutorial was a [post](https://forestry.io/blog/build-a-json-api-with-hugo/) written
+by [Régis Philibert](https://forestry.io/authors/r%C3%A9gis-philibert/). It's about doing the exact same thing in
+[Hugo](https://gohugo.io/).
+
+I hope that this solution in Jekyll is elegant, too. You just need to define some [layouts]({% link _docs/layouts.md %})
+and write a [plugin]({% link _docs/plugins.md %}) to convert your content to JSON. **Don't be afraid if you're not
+familiar with Ruby** --- the language that Jekyll is written in. It's similar to Python, and you'll get a hang of it
+quickly.
+
+By the end of the tutorial, you'll expose the following API:
+* `/` --- an index of all posts
+* `/{url}` --- details of a post
+* `/categories` --- an index of categories
+* `/categories/{category}` --- an index of posts under a category
+
+Let's dive in!
+
+{: .note .info}
+This solution needs a custom environment to support the Ruby code. This means that you can't host the site on GitHub
+Pages as-is - you'd need to use [Actions]({% link _docs/continuous-integration/github-actions.md %}) to deploy your
+site. First, see if a simpler [solution](https://gist.github.com/MichaelCurrin/f8d908596276bdbb2044f04c352cb7c7)
+created by [Michael Currin](https://github.com/MichaelCurrin) fits your needs.
+
+## Setup
+
+First, you need to go through an [installation]({% link _docs/installation.md %}) process. Having that, you can
+initialize a new Jekyll project by invoking `bundle exec jekyll new` command. Then you need to create the following
+directory structure:
+
+```
+.
+├── site
+│   ├── config.yml
+│   ├── _layouts
+│   │   ├── category.html
+│   │   └── default.html
+│   ├── _plugins
+│   │   └── api_generator.rb
+│   ├── _posts
+│   │   ├── 2020–12–20-the-first-post.md
+│   │   └── 2020–12–20-the-second-post.md
+```
+
+Unless you self-host your site, **you can rely on a generic 404 response from your server**. You no longer need `index`
+ --- you will generate a custom one later. Generating pages like `about` is out of scope of this tutorial, but in case
+you need them, they behave the same way that posts do.
+
+Under `_posts`, create two new documents. Remember that Jekyll requires you to **prepend post filename with a date**:
+
+`2020–12–20-the-first-post.md`
+
+```md
+---
+title:  "The First Post"
+category: update
+custom-property: "My custom property"
+---
+1
+```
+
+Please note that you include a `custom-property` that you will fetch later on. 
+
+`2020–12–20-the-second-post.md`
+
+```md
+---
+title:  "The Second Post"
+categories: ["update", "tutorial"]
+---
+2
+```
+
+This one belongs to two categories --- `update` and `tutorial`, and doesn't include the custom property. Because the
```

Comment:
"This one belongs to two categories  - `update` and `tutorial` - and doesn't include the custom property.